idaes/models/unit_models/gibbs_reactor.py

```diff
@@ -249,6 +249,10 @@ def gibbs_minimization(b, t, p, j):
                         j
                     ][e]
                     for e in b.control_volume.config.property_package.element_list
+                    if b.control_volume.properties_out[
+                        t
+                    ].config.parameters.element_comp[j][e]
+                    != 0
```

Comment:
Should we use 0, or a very small value? Does the Pyomo solver writer specifically check for "exactly 0" values and filter those variables out?

Reply:
This is only for exact zero (and a fixed value at that). The issue is that the solver writer is filtering these out for us, but when we use the `degrees_of_freedom` function we still see these (and hence see spurious DoF).

his is to catch cases where the multiplier is a constant of 0 and filter them out of the expression.

Reply:
I also just worked out a better solution to remove the unneeded multipliers entirely (so that they don;t end up as unused variables).